tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionSettingsTests.cs

```diff
@@ -202,7 +221,7 @@ public void With_compressors_should_return_expected_result()
             var result = subject.With(compressors: newCompressors);
 
             result.ApplicationName.Should().Be(subject.ApplicationName);
-            result.Authenticators.Should().Equal(subject.Authenticators);
+            AssertActualAuthenticatorsFactoryResult(result.AuthenticatorFactories, subject.AuthenticatorFactories);
```

Comment:
Replace with:

            result.AuthenticatorFactories.Should().Equal(subject.AuthenticatorFactories);